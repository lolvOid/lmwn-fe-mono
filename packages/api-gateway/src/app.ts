import express, { Application } from 'express';
import cors from 'cors';
import helmet from 'helmet';
import morgan from 'morgan';
import cookieParser from 'cookie-parser';
import router from './router';

const app: Application = express();

app.use(cors());
app.use(helmet());
app.use(cookieParser());
app.use(morgan('dev'));
app.use(express.json());
app.use(express.urlencoded({ extended: true }));

app.use('/', router);

export default app;
