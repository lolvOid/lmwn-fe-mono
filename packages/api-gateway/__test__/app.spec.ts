import request from 'supertest';
import app from '../src/app';

describe('Test Default Route', () => {
  test('Should return the default response', async () => {
    const res = await request(app).get(`/`);
    expect(res.text).toBe('LINE MAN Wongnai Frontend Assignment');
    expect(res.text).toMatchSnapshot();
  });
});
