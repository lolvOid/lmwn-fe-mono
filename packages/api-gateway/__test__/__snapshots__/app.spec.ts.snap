// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Default Route Should return the default response 1`] = `"LINE MAN Wongnai Frontend Assignment"`;
